### PR TITLE
Adding Database Connectivity

### LINKED ISSUE #3: Add a Database for Storing Images.
image

Suggestion: **Can Use FLASK-SQL-Alchemy**  
Currently the Server saves all the images and morphs in a local storage of the server folder itself. Implementing a database would help in optimizing speeds of different operations and also enable the path to create more feature like saving morphs for different users etc.

### PR DESCRIPTION
**Pull request by mehta-raghav**
## DATABASE SCHEMA

```
ID (PRIMARY KEY) (AUTO INCREMENT) -> INTEGER
NAME1 -> STRING(300)
NAME2 ->  STRING(300) 
IMAGE1 -> BINARY (BLOB)
IMAGE2 ->  BINARY (BLOB)
GIF -> BINARY (BLOB)     
```

#### Why this schema?
This schema takes in account for your vision of incorporating users. Creating 1 row per session will make make adding features for users easy. Just add a username and you are good to go.

#### app.py

+ Added `class IMAGES`, defining database schema.
+ Added `function image_query` querying the `IMAGE1` and `IMAGE2` from the last entry in the database.
+ Created variable `global image_array` in `function upload_file`  to keep incomplete entries going though into the database and also linking two image POST requests by the user.

#### morph.py

+ Added query script for updating row with resized image and adding gif.
+ Removed all file path dependencies. Replace them with in-memory images using cv2.

🔴 Note: I didn't find the support for binary in `imageio` module. So i converted the sequence of images into a byte blob and saved it in the gif column. So imageio still depends on creating a file for serving to the client side. But we save a blob none the less in our database.

Thanks! And feel free to contact me for any queries. I'll keep digging if I find any in-memory workarounds for gifs.

PS: You can visualize sqlite database by opening db.sqlite file [here](https://sqliteonline.com/)

### CODE REVIEW

File: morph.py [hunk 1]
```diff
@@ -18,43 +18,50 @@ def shape_to_np(shape, dtype="int"):
...
-        self.img2 = cv2.imread(image_two_path)
+    def __init__(self, img1, img2):
+        self.img1 = img1
+        self.img2 = img2
 
     def resizer(self):
         min_ht = min(self.img1.shape[0], self.img2.shape[0])
         min_wd = min(self.img1.shape[1], self.img2.shape[1])
-        dims = (min_wd, min_ht)
+        dims = (min_wd, min_ht);
```
Comment (tarunnsingh) [inline morph.py]: Semicolon!

File: app.py [hunk 2]
```diff
@@ -77,6 +120,6 @@ def morph(images):
 
 
 if __name__ == '__main__':
-    app.config['SECRET_KEY'] = os.environ.get('SECRET') or 'hyjygjgj'
+    app.config['SECRET_KEY'] = app.config['SECRET_KEY'] = os.environ.get('SECRET') or 'hyjygjgj'
```
Comment (tarunnsingh) [inline app.py]: Why this change?
```suggestion
    app.config['SECRET_KEY'] = os.environ.get('SECRET') or 'hyjygjgj'
```
Is the above line not working?

**Review by tarunnsingh — CHANGES_REQUESTED** [review]
@mehta-raghav Hey, just a few minor fixes asked here.

**Comment by tarunnsingh** [comment]
@mehta-raghav  Overall I would is say your work is absolute perfection.
I just wanted to know how is the GIF being created and is the resultant GIF's quality fine?
Could you post a sample output here.

File: morph.py (hunk 1 above)
Comment (mehta-raghav) [reply morph.py]: Whoops, typo. My bad.

File: app.py (hunk 2 above)
Comment (mehta-raghav) [reply app.py]: This was a last ditch copy and paste error, `app.config['SECRET_KEY'] = os.environ.get('SECRET') or 'hyjygjgj'` should be fine.

**Comment by mehta-raghav** [comment]
I've fixed the issues. Please re-review.

**Comment by mehta-raghav** [comment]
## Preview Gif
[](https://gifyu.com/image/6mV3)

**Review by tarunnsingh — APPROVED** [review]

**Comment by tarunnsingh** [comment]
Thank you @mehta-raghav Great Work!! 😄
